models/Employee.js

```diff
@@ -7,39 +7,70 @@ const employeeSchema = new Schema({
   email: { type: String, required: true, unique: true },
   role: { type: String, default: 'Employee' },
 
+  onboardingStatus: {
+    type: String,
+    enum: ['Not Started', 'Pending', 'Approved', 'Rejected'],
+    default: 'Not Started',
+  },
+
+  // profilePicture: { type: String, default: 'path/to/default/profile-picture.png' },
+  profilePicture: { type: String},
+
   // Onboarding and Profile Information (optional for now)
-  firstName: String,
-  lastName: String,
-  middleName: String,
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  middleName: { type: String },
+  preferredName: { type: String },
+
   address: {
     street: String,
     city: String,
     state: String,
     zip: String,
+    required:true
   },
-  phoneNumber: String,
-  workPhoneNumber: String,
-  ssn: String,
-  dob: Date,
-  gender: String,
-  workAuthorization: {
-    type: String, // e.g., "H1-B", "OPT", etc.
-    startDate: Date,
-    endDate: Date,
-    documents: [String], // list of document URLs or references
+  cellPhoneNumber: { type: String, required: true},
+  workPhoneNumber: { type: String },
+  car: {
+    make: { type: String },
+    model: { type: String },
+    color: { type: String }
+  },
+  ssn: { type: String, required: true },
+  dateOfBirth: { type: Date, required: true },
+  gender: { type: String, enum: ['Male', 'Female', 'I do not wish to answer'] },
+  citizenship: {
```

Comment:
Could use your schema to make it simpler and better. I imagine visaStatus is required field, where the rest are optional. I just wonder if we need to enforce the rule that if they are not Green Card or CItizen, they have to upload files (required). Maybe I am overthinking and it can be handled diffrently. Schema should look something like this:
  citizenship: {
    visaStatus: {type: String, required: true},
   document: { type: String}, // use this name to find the file in S3 bucket
    startDate: { type: Date },
    endDate: { type: Date } },
    optDocument: { type: mongoose.Schema.Types.ObjectId, ref:'VisaDocuments'}
  }
could see if the boolean logic is necessary later. Also, for the dates, do you think it is better if we use String instead or Date is fine?

Reply:
This schema looks good to me. 
I agree with your point. We can enforce the user to upload visa file if they are not Green Card or Citizen. I think we might apply this validation logic in UI.

For the dates, I think Date is fine.